File: server/src/main/java/com/ost/semsurf/service/DBPopulator.ex
GraphDatabaseService repository = new EmbeddedGraphDatabase("target/test-db3");
//	@Transactional
	public User createUser(User user) {				
		User user2 = repository.findByPropertyValue("name", user.getName());
		boolean isExistingUser = (user2!=null);
		log.debug("isExisting User: "+isExistingUser);
		if(!isExistingUser){	
//			template.save(user);			
			log.debug("User Created: "+user.getName());
		}else{
//			for(Page obj : user.getPages()){
//				template.createRelationshipBetween(user, obj, Link.class, "LINKED", true);
//				lrepository.save(new Link(user,page));
//				log.info(""+(null != lrepository));
//				assert(null != lrepository);
//				lrepository.save(link);
//			}
			user2.getPages().addAll(user.getPages());
//			repository.save(user2);
		}

		return user;
	}
	
	public void createPage(Set<Page> pages) {
//		for(Page page : pages){
//			prepository.save(page);
//		}
	}
	

	
//	public enum RelTypes implements RelationshipType {
//		LINKS, RELATED
//	}

GraphRepository<User>, NamedIndexRepository<User>, RelationshipOperationsRepository<User>

//		User user2 = new User("jasyon",null);
//		Page page = new Page("test.com");
//		Page page2 = new Page("test2.com");
//		Set<Page> pages = new HashSet<Page>();
//		pages.add(new Page("test.com",null));
//		user.setPages(pages);
//		pages.add(page2);
//		ArrayList<Page> pages = new ArrayList<Page>();
//		pages.toArray(user.getPages());
//		log.info("count -- "+user.getPages().size());
		
//		user2.setPages(pages);
//		prepository.save(user.getPages());
//		repository.save(user);
//		log.info("COUNT : "+user.getPages().size());
//		
//		Link link = new Link(user2, page);
//		lrepository.save(link);
//		template.createRelationshipBetween(user2, page, Link.class, "LINKED", true);
		



//	private void createUser(User user2) {
//		GraphDatabaseService repository = new EmbeddedGraphDatabase("build/test-db3");
//		Index<Node> nodeIndex = repository.index().forNodes("nodes");
//		Transaction tx = repository.beginTx();
//		try {
//			
//			String userNm = user2.getUserName();
//			Node user = nodeIndex.get("name", userNm).getSingle();
//			if(null == user){
//				log.debug("creating user: "+userNm);
//				user = repository.createNode();
//				user.setProperty("name", userNm);
//				nodeIndex.add(user, "name", userNm);
//			}
//			
//			Node link = repository.createNode();
//			link.setProperty("url", ((Page) (user2.getPages().toArray()[0])).getUrl());
//			
//			Relationship rel1 = user.createRelationshipTo(link,RelTypes.LINKS);
//			rel1.setProperty("visibility", "public");
//			
//			tx.success();
//			
//		} finally {
//			tx.finish();
//		}
//	}
	
//	@Transactional
//	public void populateDatabase(Page page) {
//		log.debug("URL: "+page.getUrl());		
//		Index<Node> nodeIndex = repository.index().forNodes("nodes");
//		//Transaction tx = repository.beginTx();
//		//try {
//			
//			String userNm = "Jayson";
//			Node user = nodeIndex.get("name", userNm).getSingle();
//			if(null == user){
//				log.debug("creating user: "+userNm);
//				user = repository.createNode();
//				user.setProperty("name", userNm);
//				nodeIndex.add(user, "name", userNm);
//			}
//			
//			Node link = repository.createNode();
//			link.setProperty("url", page.getUrl());
//			//link.setProperty("title", page.getMetaMap().get("title"));
//
//			Relationship rel1 = user.createRelationshipTo(link,RelTypes.LINKS);
//			rel1.setProperty("visibility", "public");
//			
//			//parse key words
//			String keywordString = page.getMetaMap().get("keywords");
//			if(null != keywordString){
//				String[] keywords = keywordString.split(",");			
//				for(String keyword : keywords){
//					Node kwNode = nodeIndex.get("keyword", keyword).getSingle();
//					if(null == kwNode){
//						log.debug("creating keyword: "+keyword);
//						kwNode = repository.createNode();
//						kwNode.setProperty("keyword", keyword);
//						nodeIndex.add(kwNode, "keyword", keyword);
//					}
//					Relationship rel2 = link.createRelationshipTo(kwNode,RelTypes.RELATED);
//					rel2.setProperty("visibility", "public");
//				}			
//			}
//
//			//tx.success();
//			
////		} finally {
////			tx.finish();
////		}
//	}

File: server/src/main/java/com/ost/semsurf/service/DBPopulator.java
package com.ost.semsurf.service;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import org.neo4j.graphdb.GraphDatabaseService;
import org.neo4j.graphdb.Node;
import org.neo4j.graphdb.PropertyContainer;
import org.neo4j.graphdb.Relationship;
import org.neo4j.graphdb.RelationshipType;
import org.neo4j.graphdb.Transaction;
import org.neo4j.graphdb.index.Index;
import org.neo4j.kernel.EmbeddedGraphDatabase;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.neo4j.template.Neo4jOperations;
import org.springframework.stereotype.Repository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.ost.semsurf.domain.Attribute;
import com.ost.semsurf.domain.Link;
import com.ost.semsurf.domain.Page;
import com.ost.semsurf.domain.User;
import com.ost.semsurf.repository.UserRepository;

@Service
@Transactional
public class DBPopulator {
	
	static Logger log = LoggerFactory.getLogger(DBPopulator.class);
	@Autowired UserRepository repository;
	@Autowired Neo4jOperations template;
	
	public void populateDatabase(User user) {
//		User sUser = repository.findByPropertyValue("name", user.getName());
//		boolean isExistingUser = (sUser!=null);
//		if(isExistingUser) {
//			sUser.setPages(user.getPages());
//			user = sUser;
//		}
//		repository.save(user);		
		User sUser = getUserByName(user.getName());
		boolean isExistingUser = (sUser!=null);
		if(isExistingUser) user.nodeId = sUser.nodeId; 					
		repository.save(user);
	}
	
	public User createUser(User user) {		
		return repository.save(user);		
	}
	
	public User getUserByName(String name) {		
		return repository.findByPropertyValue("name", name);	
	}
	
	public void deleteUser(User user) {		
		repository.delete(user);		
	}
	
	public boolean savePages(String name, Set<Page> pages){
		User user = getUserByName(name);
		if(null == user) return false;
		user.setPages(pages);
		repository.save(user);
		return true;
	}
	
	public Page getPage(long id){
		return template.findOne(id, Page.class);
	}

	public Attribute getAttribute(long id){
		return template.findOne(id, Attribute.class);
	}

}
